ZukoAzula2016/src/org/team1540/drivers/kauailabs/RegisterIO.java

```diff
@@ -36,8 +36,11 @@ protected void threadBody() throws InterruptedException {
                         getCurrentData();
                         Thread.sleep(1000 / update_rate_hz);
                     } catch (Throwable thr) {
-                        Logger.severe("Error in AHRS loop! Sleeping for a bit.", thr);
-                        Thread.sleep(2000);
+                        try {
+                            Logger.severe("Error in AHRS loop! " + io_provider.checkAvailable(), thr);
+                        } catch (IOException e) {
+                            Logger.severe("Should never happen", e);
+                        }
```

Comment:
"Should never happen" is wrong; "NavX did not immediately recover" might be a better error.